### PR TITLE
Add active port range configuration for FTP client factory

### PR DESCRIPTION
**Pull request by mperever**
Add configuration for FtpFileSystemConfigBuilder to set FTP active port range for [FTPClient](https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/ftp/FTPClient.html#setActivePortRange-int-int-)

### CODE REVIEW

**Comment by mperever** [comment]
Add configuration for FtpFileSystemConfigBuilder to set FTP active port range for [FTPClient](https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/ftp/FTPClient.html#setActivePortRange-int-int-)

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     * @throws IllegalArgumentException if minimum value is greater than maximum one
+     */
+    public static FtpActivePortRange of(final int minimum, final int maximum) {
+        return new FtpActivePortRange(minimum, maximum);
+    }
+
+    /**
+     * The minimum value in this range (inclusive).
+     */
+    private final int minimum;
```
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java:31]: Reuse Commons Lang Range class.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        } else {
+            final FtpActivePortRange range = (FtpActivePortRange) obj;
+            return (minimum == range.minimum) && (maximum == range.maximum);
+        }
+    }
+
+    /**
+     * Cached output hashCode (class is immutable).
+     */
+    private transient int hashCode;
```
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java:84]: Caching is not needed.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java [hunk 3]
```diff
@@ -244,6 +245,16 @@ public Boolean getPassiveMode(final FileSystemOptions options) {
         return getBoolean(options, PASSIVE_MODE);
     }
 
+    /**
+     * Gets the active port range.
+     *
+     * @param options The FileSystemOptions.
+     * @return the FTP active port range
```
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java:252]: New public methods need a 'since' tag.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @mperever 
Thank you for your PR.
The class FtpActivePortRange is not needed, just reuse the Range class from Commons Lang which is already a dependency.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java (hunk 1 above)
Comment (mperever) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java]: will do, thx

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java (hunk 3 above)
Comment (mperever) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java]: what is the value of since tag should i use ?

Comment (garydgregory) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java]: The file pom.xml says '<version>2.10.0-SNAPSHOT</version>' so use '2.10.0'

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java (hunk 2 above)
Comment (mperever) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpActivePortRange.java]: class has been removed to use generic Range

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java (hunk 3 above)
Comment (mperever) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java]: since tag is added

**Pull request merged**